docs/modules/ROOT/pages/mgmt_services/elastic_ip_manager.adoc

```diff
@@ -50,153 +50,727 @@ Note that those IP ranges should be owned by cloud provider.
-//|/ips/bulk
-//|ip state list
-//<<IP_Put1_Bulk,[sample]>>
-//
-//|Deactivate IP Address
-//|PUT
-//|/ips
-//|ip state
-//<<IP_Put2,[sample]>>
-//
-//|Deactivate IP Address Bulk
-//|PUT
-//|/ips/bulk
-//|ip state list
-//<<IP_Put2_Bulk,[sample]>>
-//
-//|Release IP State
-//|DELETE
-//|/ips/{range_id}/{ip}
-//|ip state
-//<<IP_Delete,[sample]>>
-//
-//|Release IP State Bulk
-//|DELETE
-//|/ips/bulk
-//|ip state list
-//<<IP_Delete_Bulk,[sample]>>
-//
-//|Create IP Range
-//|POST
-//|/ips/range
-//|ip range
-//<<IP_Range_Post,[sample]>>
-//
-//|Get IP Range
-//|GET
-//|/ips/range/{range_id}
-//|ip range
-//<<IP_Range_Get,[sample]>>
-//
-//|List all IP Range
-//|GET
-//|/ips/range
-//|ip range list
-//<<IP_Range_List,[sample]>>
+|Create an elastic IP
+|POST
+|/project/{project_id}/elasticips
+|elasticip info
+<<Elastic_IP_Create,[sample]>>
+
+|Get the elastic IP
+|GET
+|/project/{project_id}/elasticips/{elasticip_id}
+|elasticip info
+<<Elastic_IP_Get,[sample]>>
+
+|Get all elastic IPs
+|GET
+|/project/{project_id}/elasticips
+|elasticip info list
+<<Elastic_IP_Get_List,[sample]>>
+
+|Delete the elastic IP
+|DELETE
+|/project/{project_id}/elasticips/{elasticip_id}
+|
+<<Elastic_IP_Delete,[sample]>>
+
+|Bulk create elastic IPs
+|POST
+|/project/{project_id}/elasticips
+|elasticip info list
+<<Elastic_IP_Bulk_Create,[sample]>>
+
+|Bulk delete elastic IPs
+|DELETE
+|/project/{project_id}/elasticips
+|
+<<Elastic_IP_Bulk_Delete,[sample]>>
+
+|Update the elastic IP
+|PUT
+|/project/{project_id}/elasticips/{elasticip_id}
+|elasticip info
+<<Elastic_IP_Update,[sample]>>
+
+|Create an elastic IP range
+|POST
+|/elasticip-ranges
+|elasticip range info
+<<Elastic_IP_Range_Create,[sample]>>
+
+|Get the elastic IP range
+|GET
+|/elasticip-ranges/{elasticip_range_id}
+|elasticip range info
+<<Elastic_IP_Range_Get,[sample]>>
+
+|Get all elastic IP ranges
+|GET
+|/elasticip-ranges
+|elasticip range info list
+<<Elastic_IP_Range_Get_List,[sample]>>
+
+|Delete the elastic IP range
+|DELETE
+|/elasticip-ranges/{elasticip_range_id}
+|
+<<Elastic_IP_Range_Delete,[sample]>>
+
+|Update the elastic IP range
+|PUT
+|/project/{project_id}/elasticip-ranges/{elasticip_range_id}
+|elasticip range info
+<<Elastic_IP_Range_Update,[sample]>>
+
 |===
 
 === API Specification
 
-NOTE: The follows is an example from link:private_ip_manager.adoc[private IP manager].
+anchor:Elastic_IP_Create[]
+**(1) Create an elastic IP**
+
+* Method: `POST`
+
+* Request: `/project/{project_id}/elasticips`
+
+* Request Parameter: `@PathVariable String project_id, @RequestBody ElasticIpStateJson elasticip`
+
+* Response: elastic ip info
+* Normal response codes: 200
+* Error response codes: 400, 412, 500
+
+* Example
+
+....
+Request:
+http://127.0.0.1:8080/elasticips
+
+Body:
+{
+    "elasticip":
+    {
+        "id": null,
+        "project_id":"11223344-5566-7788-990011223300",
+        "name":"eip1",
+        "description":"elastic ip 1",
+        "range_id":"11223344-5566-7788-990011223301",
```

Comment:
Is range_id an optional or a required parameter here?

Reply:
you are right, in mostly scenarios there is only one existing elasticip range. I will change the range_id to optional

Reply:
Cool thanks!